pcsx2/gui/Dialogs/ModalPopups.h

```diff
@@ -18,56 +18,352 @@
+		{
+			GSType id;
+			std::unique_ptr<char[]> data;
+			int length;
+			GSTransferPath path;
+		};
+		enum ButtonState
+		{
+			Step,
+			RunCursor,
+			RunVSync
+		};
+		static constexpr const char* GSTransferPathNames[256] = {
+			"Path1Old",
+			"Path2",
+			"Path3",
+			"Path1New",
+			"Dummy"
+		};
+		enum GifFlag : u8
+		{
+			GIF_FLG_PACKED = 0,
+			GIF_FLG_REGLIST = 1,
+			GIF_FLG_IMAGE = 2,
+			GIF_FLG_IMAGE2 = 3
+		};
+		static constexpr const char* GifFlagNames[256] = {
+			"GIF_FLG_PACKED",
+			"GIF_FLG_REGLIST",
+			"GIF_FLG_IMAGE",
+			"GIF_FLG_IMAGE2"
+		};
+		static constexpr const char* GsPrimNames[256] = {
+			"GS_POINTLIST",
+			"GS_LINELIST",
+			"GS_LINESTRIP",
+			"GS_TRIANGLELIST",
+			"GS_TRIANGLESTRIP",
+			"GS_TRIANGLEFAN",
+			"GS_SPRITE",
+			"GS_INVALID"
+		};
+		static constexpr const char* GsIIPNames[256] = {
+			"FlatShading",
+			"Gouraud"
+		};
+		static constexpr const char* GsFSTNames[256] = {
+			"STQValue",
+			"UVValue"
+		};
+		static constexpr const char* GsCTXTNames[256] = {
+			"Context1",
+			"Context2"
+		};
+		static constexpr const char* GsFIXNames[256] = {
+			"Unfixed",
+			"Fixed"
+		};
+		static constexpr const char* TEXTCCNames[256] = {
+			"RGB",
+			"RGBA"
+		};
+		static constexpr const char* TEXTFXNames[256] = {
+			"MODULATE",
+			"DECAL",
+			"HIGHLIGHT",
+			"HIGHLIGHT2"
+		};
+		static constexpr const char* TEXCSMNames[256] = {
+			"CSM1",
+			"CSM2"
+		};
+		// a GNU extension exists to initialize array at given indices which would be 
+		// exactly what we need here but, obviously, MSVC is at it again to make our 
+		// life harder than sandpaper on your skin, so we make do
+		// clang-format off
+		static constexpr const char* TEXCPSMNames[256] = {
+			"PSMCT32",
+			"",
+			"PSMCT16",
+			"","","","","","","",
+			"PSMCT16S"
+		};
+		static constexpr const char* TEXPSMNames[256] = {
+			"PSMCT32",
+			"PSMCT24",
+			"PSMCT16",
+			"","","","","","","",
+			"PSMCT16S",
+			"","","","","","","","",
+			"PSMT8",
+			"PSMT4",
+			"","","","","","",
+			"PSMT8H",
+			"","","","","","","","",
+			"PSMT4HL",
+			"","","","","","","",
+			"PSMT4HH",
+			"","","",
+			"PSMZ32",
+			"PSMZ24",
+			"PSMZ16",
+			"","","","","","","",
+			"PSMZ16S"
+		};
+		// clang-format on
+
+		// the actual type is defined above thanks to preprocessing magic
+		enum GSDUMP_GIFREG_NAME : GSDUMP_GIFREG_TYPE
+		{
+#define X(name, value) name = value,
+			GSDUMP_GIFREG(X)
+#undef X
+		};
+		constexpr auto GIFRegName(GSDUMP_GIFREG_NAME e) noexcept
+		{
+#define X(name, value)               \
+	case (GSDUMP_GIFREG_NAME::name): \
+		return #name;
+			switch (e)
+			{
+				GSDUMP_GIFREG(X)
+			}
+#undef X
+			return "UNKNOWN";
+		};
+		struct GSEvent
+		{
+			ButtonState btn;
+			int index;
+		};
+		std::vector<GSEvent> m_button_events;
+		std::vector<GSData> m_dump_packets;
+
+		float m_stored_q = 1.0;
+		void ProcessDumpEvent(GSData& event, char* regs);
+		void GenPacketList(std::vector<GSData>& dump);
```

Comment:
Assuming that the only times that this is currently used in the code will remain that way, you don't need `std::vector<GSData>& dump` as a parameter. When it gets used on lines 580 & 678, you end up passing the m_dump_packets member variable regardless.